### PR TITLE
Clustering and Wall Detection

### PR DESCRIPTION
**Pull request by jhbruhn**
The cluster algorithm (DBSCAN) has been implemented and uses the values from the sensor fusion node to generate clusters. The parameters of the DBSCAN algorithm are configureable via dynamic reconfigure.
In addition to that, a primitive wall_detection algorithm has been implemented which chooses the clusters furthest away to the left and right in a configurable radius around the car. This obviously has some flaws but works pretty okay during initial testing.

### CODE REVIEW

File: ros_ws/src/autonomous/wall_detection/src/wall_detection.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    // clean up
+    for (auto itr = clustersUsed.begin(); itr != clustersUsed.end(); ++itr)
+        delete itr->second;
+}
+
+void WallDetection::publishObstacles(std::unordered_map<int, std::vector<pcl::PointXYZI>*> mapClusters,
+                                     std::pair<int, int> wallIDs)
+{
+    pcl::PointCloud<pcl::PointXYZI>::Ptr msg(new pcl::PointCloud<pcl::PointXYZI>);
```
Comment (Rothu) [inline ros_ws/src/autonomous/wall_detection/src/wall_detection.cpp:74]: Potential memory leak

File: ros_ws/src/autonomous/wall_detection/src/wall_detection.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            }
+        }
+    }
+    pcl_conversions::toPCL(ros::Time::now(), msg->header.stamp);
+    m_obstacles_publisher.publish(msg);
+}
+
+void WallDetection::publishWall(std::vector<pcl::PointXYZI>* wallLeft, std::vector<pcl::PointXYZI>* wallRight)
+{
+    pcl::PointCloud<pcl::PointXYZI>::Ptr msg(new pcl::PointCloud<pcl::PointXYZI>);
```
Comment (Rothu) [inline ros_ws/src/autonomous/wall_detection/src/wall_detection.cpp:94]: memory leak

**Review by Rothu — APPROVED** [review]

**Review by ghost — APPROVED** [review]
i agree

**Comment by ghost** [comment]
(values["straight_length"] * 4 + Math.PI * 2 * values["curve_radius"]).toFixed(2);

**Pull request merged**